catalyst_voices/packages/internal/catalyst_voices_repositories/test/src/dto/document/document_data_dto_test.dart

```diff
@@ -103,6 +99,119 @@ void main() {
+            },
+            'campaignId': {
+              'id': campaignId,
+              'ver': campaignId,
+              'type': 'signed',
+            },
+          };
+
+          final campaignRef = DocumentRefDto(
+            id: campaignId,
+            ver: campaignId,
+            type: DocumentRefDtoType.signed,
+          );
+          final brandRef = DocumentRefDto(
+            id: brandId,
+            ver: brandId,
+            type: DocumentRefDtoType.signed,
+          );
+
+          // When
+          final dto = DocumentDataMetadataDto.fromJson(oldJson);
+
+          // Then
+          expect(dto.parameters.contains(campaignRef), isTrue);
+          expect(dto.parameters.contains(brandRef), isTrue);
+        });
+
+        test('do nothing when migration not needed', () {
+          // Given
+          final json = <String, dynamic>{
+            'type': DocumentType.proposalDocument.uuid,
+            'selfRef': {
+              'id': DocumentRefFactory.randomUuidV7(),
+              'version': DocumentRefFactory.randomUuidV7(),
+              'type': 'signed',
+            },
+            'parameters': [
+              {
+                'id': DocumentRefFactory.randomUuidV7(),
+                'version': DocumentRefFactory.randomUuidV7(),
+                'type': 'signed',
+              },
+              {
+                'id': DocumentRefFactory.randomUuidV7(),
+                'version': DocumentRefFactory.randomUuidV7(),
+                'type': 'signed',
+              },
+            ],
+          };
+
+          // When
+          final migrated = DocumentDataMetadataDto.migrateJson1(json);
+
+          // Then
+          expect(identical(json, migrated), isTrue);
+        });
+      });
+
+      group('4', () {
+        test('version and id migration works as expected', () {
+          // Given
+          final oldJson = <String, dynamic>{
+            'type': DocumentType.proposalDocument.uuid,
+            'selfRef': {
+              'id': DocumentRefFactory.randomUuidV7(),
+              'version': DocumentRefFactory.randomUuidV7(),
+              'type': 'signed',
+            },
+          };
+
+          // When
+          final dto = DocumentDataMetadataDto.fromJson(oldJson);
+
+          // Then
+          expect(dto.contentType, DocumentContentType.json);
+        });
+
+        test('do nothing when migration not needed', () {
+          // Given
+          final json = <String, dynamic>{
+            'contentType': DocumentContentType.json.value,
+            'type': DocumentType.proposalDocument.uuid,
+            'selfRef': {
+              'id': DocumentRefFactory.randomUuidV7(),
+              'version': DocumentRefFactory.randomUuidV7(),
+              'type': 'signed',
+            },
+          };
+
+          // When
+          final migrated = DocumentDataMetadataDto.migrateJson1(json);
+
+          // Then
+          expect(identical(json, migrated), isTrue);
```

Comment:
There's a custom matcher for it:

```suggestion
          expect(migrated, same(json));
```